src/commands/deploy.ts

```diff
@@ -47,27 +47,22 @@ export async function deploy(tree: AzureTreeDataProvider, outputChannel: vscode.
 async function getJavaFolderPath(outputChannel: vscode.OutputChannel, basePath: string, ui: IUserInterface): Promise<string> {
     outputChannel.show();
     await cpUtils.executeCommand(outputChannel, basePath, 'mvn', 'clean', 'package', '-B');
-    const targetFolder: string = path.join(basePath, 'target', 'azure-functions');
-    if (!await fse.pathExists(targetFolder)) {
-        throw new NoPackagedJavaFunctionError();
-    }
-    const packagedFolders: string[] = fse.readdirSync(targetFolder);
-    if (packagedFolders.length === 0) {
-        throw new NoPackagedJavaFunctionError();
-    } else if (packagedFolders.length === 1) {
-        return path.join(targetFolder, packagedFolders[0]);
+    const pomLocation: string = path.join(basePath, 'pom.xml');
+    const functionAppName: string | undefined = await getFunctionAppNameInPom(pomLocation);
+    const targetFolder: string = functionAppName ? path.join(basePath, 'target', 'azure-functions', functionAppName) : '';
+    if (functionAppName && await fse.pathExists(targetFolder)) {
+        return targetFolder;
     } else {
-        return path.join(targetFolder, await promptForPackagedFolder(ui, packagedFolders));
+        const message: string = localize('azFunc.cannotFindPackageFolder', 'Cannot find the packaged function folder, would you like to specify the folder location?');
```

Comment:
Since you're fixing the promise bug below anyways, can you leverage DialogResponse.cancel here (it was just merged to master)

Reply:
Sure